File: Source/Affixes/Stealth.cs
﻿public class Stealth : Affix
{
    private const int AFFIX_ABILITY = Constants.ABILITY_GHOSTAFFIX;

    public Stealth(Wolf unit) : base(unit)
    {
        Name = $"{Colors.COLOR_GREY}Stealth|r";
    }

    public override void Apply()
    {
        Unit.Unit.AddAbility(AFFIX_ABILITY);
        base.Apply();
    }

    public override void Remove()
    {
        Unit.Unit.RemoveAbility(AFFIX_ABILITY);
        base.Remove();
    }

    public override void Pause(bool pause)
    {
        // no logic needed atm
    }

}

File: src/Affixes/Stealth.ts
export class Stealth extends Affix {
    private AFFIX_ABILITY: number = Constants.ABILITY_GHOSTAFFIX

    public constructor(unit: Wolf) {
        super(unit)
        name = '{Colors.COLOR_GREY}Stealth|r'
    }

    public override Apply() {
        this.Unit.Unit.addAbility(this.AFFIX_ABILITY)
        this.Unit.Unit.addAbility(this.AFFIX_ABILITY)
        super.Apply()
    }

    public override Remove() {
        this.Unit.Unit.removeAbility(this.AFFIX_ABILITY)
        super.Remove()
    }

    public override Pause(pause: boolean) {
        // no logic needed atm
    }
}
